File: C#/프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.cs
using System;
using System.Collections.Generic;

public class Solution {
    List<string> jokaBabb = new List<string>() { "aya", "ye", "woo", "ma" };
    
    private bool checkCanSpeak(string word) {
        if (word.Length == 0) return true;
        foreach(var j in jokaBabb) {
            if (word.StartsWith(j)) {
                return checkCanSpeak(word.Substring(j.Length));
            }
        }
        return false;
    }
    
    public int solution(string[] babbling) {
        int answer = 0;
        foreach(var b in babbling) {
            if (checkCanSpeak(b)) answer++;
        }
        return answer;
    }
}

File: JavaScript/프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.js
const jokaBabb = ["aya", "ye", "woo", "ma"];

function checkCanSpeak(word) {
    for (const j of jokaBabb) {
        if (word.length == 0)
            return true;
        if (word.startsWith(j))
            return checkCanSpeak(word.substring(j.length));
    }
    return false
}

function solution(babbling) {
    let answer = 0;
    babbling.forEach((b) => {
        if (checkCanSpeak(b)){
            answer += 1;
        }
    })
    return answer;
}